src/Commands/SettingsCommandGroup.cs

```
@@ -85,37 +85,48 @@ public class SettingsCommandGroup : CommandGroup {
-            .WithFooter(footer.ToString())
-            .Build();
+        const int optionsPerList = 10;
+        var totalPages = (AllOptions.Length + optionsPerList - 1)/optionsPerList;
+        var firstOptionOnPage = optionsPerList * page - optionsPerList;
+        var lastOptionOnPage = optionsPerList * page - 1;
 
-        if (optionsPerList * page - optionsPerList >= AllOptions.Length) {
-            embed = new EmbedBuilder().WithSmallTitle(Messages.PageNotFound, currentUser)
-                .WithDescription($"{Messages.PagesAllowed}: {Markdown.Bold(totalPages.ToString())}")
+        if (firstOptionOnPage >= AllOptions.Length) {
+            var embed = new EmbedBuilder().WithSmallTitle(Messages.PageNotFound, currentUser)
+                .WithDescription(string.Format(Messages.PagesAllowed, Markdown.Bold(totalPages.ToString())))
                 .WithColour(ColorsList.Red)
                 .Build();
-        }
 
-        return await _feedbackService.SendContextualEmbedResultAsync(embed, ct);
+            return await _feedbackService.SendContextualEmbedResultAsync(embed, ct);
+        } else {
+            footer.Append($"{Messages.Page} {page}/{totalPages} ");
+            for (var i = 1; i <= totalPages; i++) footer.Append(i == page ? "●" : "○");
+
+            for (var i = firstOptionOnPage; i <= lastOptionOnPage; i++) {
+                try {
+                    var optionName = AllOptions[i].Name;
+                    var optionValue = AllOptions[i].Display(cfg);
+
+                    description.AppendLine($"- {$"Settings{optionName}".Localized()}")
+                        .Append(" - ")
+                        .Append(Markdown.InlineCode(optionName))
+                        .Append(": ")
+                        .AppendLine(optionValue);
+                } catch (IndexOutOfRangeException) {
+                     // Ignoring IndexOutOfRangeException in order to continue showing the list of settings if the options are over
+                }
```

Boyfriend is a project that inherits Remora's usage of Results for error handling. No exception should EVER be thrown in ANY normal operation. If you're encountering IOORE, you need to correct your code, not swallow the exception (it's code smell and a performance hit)